docs/SemanticARC.md

```diff
@@ -283,7 +337,11 @@ This is correct. But what if we want to get rid of the destroy_value by performi
       return %result : $()
     }
 
-The key reason to have the guarantee_lifetime/destroy_lifetime_guarantee is that it encapsulates via the use-def list the region where the lifetime of the object is guaranteed. Once this is done, we then perform the @owned -> @guaranteed optimization [[3]](#footnote-3):
+The key reason to have the guarantee_lifetime/destroy_lifetime_guarantee is that it encapsulates via the use-def list the region where the lifetime of the object is guaranteed.
+
+[EE] I don't understand why we need guarantee_lifetime/destroy_lifetime_guarantee. Seems to me that even without that the value can be proved to be guaranteed in that region.
```

Comment:
I was showing how one can use local signature optimizations for optimization. I could be smarter by looking for a guaranteed parameter on the same RC identity. I.e. that one can go bottom up through scopes, changing them from +0 to +1. And then once all lifetimes across all scopes is +0, one can just run the current ARC sequence opts.